Separate map setup from robits.js

tnunamak opened the pull request:
More to come.

Commit:
Separate map setup from robits.js

chrispound commented:
saw 2 issues: see 
https://github.com/Atlas-/robits/commit/d499cf50185069439e7025e1b1922509051cd571

If this works for you then this should be good to merge!

Commit:
portalTiles where not defined. needed to get the reference from map
player was colliding with checkpoint after game reset. added movement clear if a player won the game.
